packages/core/src/compiler.ts

```diff
@@ -303,21 +309,28 @@ export default class Compiler {
     }
   }
 
-  setFormatter(key: string) {
+  setFormatter(key: string, parentKey?: string) {
     if (this.runtimeIncludes(key, 'formatter')) return;
-    let cf = this.options.customFormatters[key];
+    const cf = this.options.customFormatters[parentKey || key];
     if (cf) {
-      if (typeof cf === 'function') cf = { formatter: cf };
+      const cfo = typeof cf === 'function' ? { formatter: cf } : cf;
+
       this.runtime[key] = Object.assign(
-        cf.formatter,
+        cloneFunction(cfo.formatter),
```

Comment:
Ok, I see now why mutating the formatter can be a bad idea: When it's not a built-in formatter that we're providing, it means we're modifying a value that a user has provided.

So using `.bind()` here is appropriate. See below for why you ought to inline the code for that, as you had it previously.